deliver/lib/deliver/upload_metadata.rb

```diff
@@ -451,10 +459,43 @@ def retry_if_nil(message, tries: 5, wait_time: 10)
       end
     end
 
-    # Finding languages to enable
-    def verify_available_info_languages!(options, app, languages)
-      app_info = fetch_edit_app_info(app)
+    # Checking if the metadata to update includes localised App Info
+    def updating_localised_app_info?(options, app, app_info)
+      app_info ||= fetch_live_app_info(app)
```

Comment:
Wait, after browsing the codebase a bit, I just discovered that [`Spaceship::ConnectAPI::App` has an `fetch_latest_app_info` method](https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/connect_api/models/app.rb#L150-L154) in addition to [the more-specific `fetch_edit_app_info` and `fetch_live_app_info` ones that filter the results according to the app state](https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/connect_api/models/app.rb#L116-L148).

I wonder if it wouldn't make sense to use that `fetch_latest_app_info directly then, instead of doing the fallback manually here? 🤔 Though maybe that wouldn't have the same fallback behavior as the one you're using here… 🤔  wdyt?

Reply:
Good catch! I wasn't aware of that API. But, if I understand it right, this is equivalent (though a better solution) to my initial approach of using `fetch_edit_app_info(app) || fetch_live_app_info(app)` which had the problem that it was then difficult to properly give feedback to the user and to understand when we could progress and when it's better to fail here. 
We added manual fallback to address this problem. 
Does it make sense to you?

Reply:
Yeah good point. That's also what I feared and why I asked about if it'd provide the same fallback behavior and all.

Better to give us the occasion to provide nicer error reporting to our users I think, and thus to keep your implementation to be able to distinguish both cases 👍